AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import SQLite
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+@testable import AWSPluginsCore
+
+class MutationEventExtensionsTest: BaseDataStoreTests {
+    /// - Given: A create mutationSync with a given version
+    /// - When: Mutation event table contains update and delete mutation events added in order, having same model id
+    /// - Then: Update mutation event should be updated with version of create mutationSync
+
+    func testQueryAfterUpdatePendingMutationEventVersion() {
```

Comment:
instead of jumping straight into a complex test case, have a simple one first. just save one mutation event with nil version, and then call your method to update it. 

1. save a mutation event with nil version
2. check that head of the queue contains your mutation event with nil version
3. call you method with expected version
4. check the head of the queue contains your mutation event with expected version